### PR TITLE
Use a default user agent

### LINKED ISSUE #500: Create unique user-agent
Currently, we use the standard go-http-lib user agent. However, to be able to better track the usage of the tool, we should use our own user-agent string, e.g., `"<program> <version>"`.

### PR DESCRIPTION
**Pull request by koplas**
Closes #500

### CODE REVIEW

File: util/client.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ func (hc *HeaderClient) Do(req *http.Request) (*http.Response, error) {
 			req.Header.Add(key, v)
 		}
 	}
+
+	// Use default user agent if none is set
+	if userAgent := hc.Header.Get("User-Agent"); userAgent == "" {
+		req.Header.Add("User-Agent", "csaf-distribution "+SemVersion)
```
Comment (bernhardreiter) [inline util/client.go:67]: The generated user agent must follow https://datatracker.ietf.org/doc/html/rfc9110#name-user-agent which it currently does not.

**Review by bernhardreiter — CHANGES_REQUESTED** [review]
* Documentation changes are missing.
* The chosen user-agent does not seem standard compliant (see comment).

**Review by JanHoefelmeyer — APPROVED** [review]
LGTM

**Review by bernhardreiter — APPROVED** [review]
LGTM
@JanHoefelmeyer can you also do a functional test, please.

**Comment by JanHoefelmeyer** [comment]
Have done a functional test, so LGTM & works

**Review by JanHoefelmeyer — APPROVED** [review]

**Pull request merged**